engine/dlib/src/dlib/job_thread.cpp

```diff
@@ -152,14 +155,12 @@ void Destroy(HContext context)
         return;
 
 #if defined(DM_HAS_THREADS)
-    dmAtomicStore32(&context->m_ThreadContext.m_Run, 0);
     {
         DM_MUTEX_SCOPED_LOCK(context->m_ThreadContext.m_Mutex);
-        // Wake up the worker threads so it can exit and allow us to join
-        for (int i = 0; i < context->m_Threads.Size(); ++i)
-        {
-            dmConditionVariable::Signal(context->m_ThreadContext.m_WakeupCond);
-        }
+
+        context->m_ThreadContext.m_Run = false;;
```

Comment:
Now we know that the Run and Wait() are within the same protected scope on the thread and here as well.